worker/src/error.rs

```diff
@@ -1,19 +1,70 @@
+use crate::Response;
+use serde_json::json;
+use std::result::Result;
+use thiserror::Error;
 use wasm_bindgen::{JsCast, JsValue};
 
 /// All possible Error variants that might be encountered while working with a Worker.
-#[derive(Debug)]
+
+#[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum Error {
     BadEncoding,
     BodyUsed,
+    ResponseError(Response),
     Json((String, u16)),
     JsError(String),
     Internal(JsValue),
     BindingError(String),
-    RouteInsertError(matchit::InsertError),
+    RouteInsertError(#[from] matchit::InsertError),
     RouteNoDataError,
-    RustError(String),
-    SerdeJsonError(serde_json::Error),
+    ParseError(#[from] url::ParseError),
+    StringError(String),
+    SerdeJsonError(#[from] serde_json::Error),
+    RustError(#[from] anyhow::Error),
+}
+
+#[inline]
+pub fn unwrap_abort<T, E>(o: Result<T, E>) -> T {
+    use std::process;
+    match o {
+        Ok(t) => t,
+        Err(_) => process::abort(),
+    }
+}
+
+impl From<Error> for Response {
+    fn from(e: Error) -> Self {
+        match e {
+            Error::ResponseError(resp) => resp,
+            Error::Json((msg, code)) => {
+                let res = Response::from_json(&json!({
+                    "msg": msg,
+                    "code": code
+                }))
+                .map(|r| r.with_status(code));
+                unwrap_abort(res)
+            }
+            _ => unwrap_abort(Response::error("An error occurred.", 500)),
+        }
+    }
+}
+
+impl From<Error> for worker_sys::Response {
+    fn from(e: Error) -> Self {
+        match e {
```

Comment:
This seems like it should just convert the error to a `Response` then `.into()` it into a sys response.

Reply:
Agreed, fixed